packages/notifier/src/marshal.js

```diff
@@ -0,0 +1,104 @@
+// @ts-check
+import { E } from '@endo/eventual-send';
+import { Far, makeMarshal } from '@endo/marshal';
+import { observeIteration } from './asyncIterableAdaptor.js';
+
+/** @template T @typedef {import('@endo/marshal').CapData<T>} CapData */
+/** @template T @typedef {import('@endo/eventual-send').EOnly<T>} EOnly */
+
+/**
+ * @typedef {{
+ *   [index: string]: any,
+ *   unserializer: ERef<{
+ *     unserialize: import('@endo/marshal').Unserialize<unknown>
+ *   }>
+ * }} SubscriberDetails information and capabilities needed to follow a
+ * publisher out-of-band
+ */
+
+/**
+ * @template T
+ * @typedef {object} Publisher
+ * @property {(obj: ERef<T>) => void} publish publishes a jsonable value
```

Comment:
What do you mean by "jsonable"? If you mean what I expect, why do you hate bigints ;)

Reply:
Because the publisher publishes only marshalled values, it isn't required to accept anything but jsonable input.